File: calculus_and_numerical_methods/trapezoidal_rule.py
"""
trapezoidal_rule.py
Numerical integration using the Trapezoidal Rule.
Author: Scott
"""

def f(x):
    return x**2  # Example: integrate x^2

def trapezoidal(a, b, n):
    h = (b - a) / n
    result = 0.5 * (f(a) + f(b))
    for i in range(1, n):
        result += f(a + i * h)
    return result * h

if __name__ == "__main__":
    a = float(input("Lower bound a (default 0): ") or 0)
    b = float(input("Upper bound b (default 1): ") or 1)
    n = int(input("Number of intervals (default 100): ") or 100)
    print(f"Integral ≈ {trapezoidal(a, b, n):.6f}")

File: calculus_and_numerical_methods/trapezoidal_rule.cpp

/**
 * Trapezoidal Rule for Numerical Integration
 * Author: Scott Douglass
 * License: MIT
 */

#include <iostream>
#include <cmath>

double f(double x) {
    return std::sin(x);  // Example function
}

double trapezoidal(double a, double b, int n) {
    double h = (b - a) / n;
    double result = 0.5 * (f(a) + f(b));
    for (int i = 1; i < n; ++i)
        result += f(a + i * h);
    return result * h;
}

int main() {
    std::cout << "Approx ∫sin(x) from 0 to π ≈ " << trapezoidal(0, M_PI, 100) << std::endl;
    return 0;
}
